manim/mobject/geometry/boolean_ops.py

```diff
@@ -51,13 +53,14 @@ def _convert_2d_to_3d_array(
         >>> a = _BooleanOps()
         >>> p = [(1, 2), (3, 4)]
         >>> a._convert_2d_to_3d_array(p)
-        [array([1., 2., 0.]), array([3., 4., 0.])]
+        array([[1., 2., 0.],
+               [3., 4., 0.]])
         """
-        points = list(points)
-        for i, point in enumerate(points):
+        list_of_points = list(points)
+        for i, point in enumerate(list_of_points):
             if len(point) == 2:
-                points[i] = np.array(list(point) + [z_dim])
-        return points
+                list_of_points[i] = np.array(list(point) + [z_dim])
+        return np.array(list_of_points)
```

Comment:
We can probably use `np.asarray` to make sure we avoid making unnecessary copies.

Reply:
Good idea. It works fine according to mypy.